web-console/future-openshift/bulk-actions/bulk-actions.md

```diff
@@ -0,0 +1,38 @@
+# Bulk actions
+
+Bulk actions will be added to a number of pages in the OpenShift console.
+
+
+## Behavior
+* The bulk selector will be the first component in the toolbar on pages where bulk actions are available.
+* Users can click the checkbox in the bulk selector to select all items in the list. If all items are selected, the bulk selector checkbox will be filled with a check and the bulk selector will read 'All items selected'. Clicking the bulk selector checkbox again will deselect all items.
+* Users can also choose to select a group of items by clicking the associated checkboxes next to them. The bulk selector checkbox will be filled in with a '-' to indicate that some items are selected, and the bulk selector will display the number of items selected (e.g. '2 selected'). Clicking the bulk selector checkbox will deselect all items.
+* If no items are selected, the bulk actions dropdown will be enabled, but the actions within will be disabled. A tooltip will explain to users that they must select resources in order to take actions.
+
+![1](img/1.png)
+
+![2](img/2.png)
+
+![3](img/3.png)
+
+![4](img/4.png)
+
+## Actions
+* Only specific actions will be available to take on multiple resources. The default bulk actions are 'Add Labels', 'Add Annotations', and 'Delete Resources'.
+* Additional actions may be added in future depending on the resource type.
+
+### Add Labels
+![5](img/add-labels.png)
```

Comment:
I like the intention of listing the resources that have been selected here that will be affected, but just curious how this might scale. Should we say there is a limit of how many we actually list out just in case the user has chosen 20 resources or so where listing them out by name would get long?

Reply:
That's a really good point. Maybe we list it out up until 4 or 5 resources? What do you think? I'm wondering if after we cap it should we just say "Adding labels to 20 resources" or should we also include some way to expand the area to see which resources? I'm leaning toward the former, but curious your thoughts!

Reply:
Yeah I think I've seen a good limit be 5 and then you either do a fancy expansion to show all in a list or just do the "20 resources will be updated" type thing :) I think the latter is a good starting point, so maybe as easy as just noting it in the markdown rather than even needing a whole extra wireframe.